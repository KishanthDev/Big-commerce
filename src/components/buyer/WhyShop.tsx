export default function WhyShop() {
  const features = [
    {
      icon: "üîç",
      title: "Unique Products",
      description:
        "Discover one-of-a-kind items you won't find in big box stores. Every product has a story.",
    },
    {
      icon: "ü§ù",
      title: "Support Local",
      description:
        "Your purchases directly support independent creators and small business owners in your community.",
    },
    {
      icon: "‚≠ê",
      title: "Verified Quality",
      description:
        "Every seller is vetted for quality and service. Shop with confidence knowing we stand behind them.",
    },
  ];

  return (
    <section className="py-16 dark:bg-black">
      <div className="container mx-auto px-5">
        <h2 className="text-3xl font-semibold dark:text-white text-center mb-12">
          Why Shop With Us?
        </h2>
        <div className="grid grid-cols-1 md:grid-cols-3 gap-6">
          {features.map((feature, index) => (
            <div
              key={index}
              className="bg-white dark:bg-gray-300 p-8 rounded-lg shadow-sm text-center"
            >
              <div className="bg-blue-50 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-5 text-blue-600 text-2xl">
                {feature.icon}
              </div>
              <h3 className="text-xl dark:text-black font-semibold mb-3">{feature.title}</h3>
              <p className="text-gray-600 text-sm">{feature.description}</p>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
}
