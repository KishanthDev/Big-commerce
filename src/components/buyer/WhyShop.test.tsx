import { render, screen } from "@testing-library/react";
import WhyShop from "./WhyShop";

describe("WhyShop Component", () => {
  const mockFeatures = [
    {
      icon: "🔍",
      title: "Unique Products",
      description:
        "Discover one-of-a-kind items you won't find in big box stores. Every product has a story.",
    },
    {
      icon: "🤝",
      title: "Support Local",
      description:
        "Your purchases directly support independent creators and small business owners in your community.",
    },
    {
      icon: "⭐",
      title: "Verified Quality",
      description:
        "Every seller is vetted for quality and service. Shop with confidence knowing we stand behind them.",
    },
  ];

  test("displays the main heading", () => {
    render(<WhyShop />);
    expect(
      screen.getByRole("heading", { name: /Why Shop With Us?/i }),
    ).toBeInTheDocument();
  });

  test("displays correct feature information", () => {
    render(<WhyShop />);

    mockFeatures.forEach((feature) => {
      expect(screen.getByText(feature.icon)).toBeInTheDocument();
      expect(screen.getByText(feature.title)).toBeInTheDocument();
      expect(screen.getByText(feature.description)).toBeInTheDocument();
    });
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<WhyShop />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("has proper accessibility attributes", () => {
    render(<WhyShop />);
    expect(screen.getByRole("heading", { level: 2 })).toHaveTextContent(
      "Why Shop With Us?",
    );
    expect(screen.getAllByRole("heading", { level: 3 })).toHaveLength(
      mockFeatures.length,
    );
  });
});
