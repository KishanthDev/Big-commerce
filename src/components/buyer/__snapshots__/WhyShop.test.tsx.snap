// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhyShop Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="py-16 dark:bg-black"
  >
    <div
      class="container mx-auto px-5"
    >
      <h2
        class="text-3xl font-semibold dark:text-white text-center mb-12"
      >
        Why Shop With Us?
      </h2>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-6"
      >
        <div
          class="bg-white dark:bg-gray-300 p-8 rounded-lg shadow-sm text-center"
        >
          <div
            class="bg-blue-50 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-5 text-blue-600 text-2xl"
          >
            üîç
          </div>
          <h3
            class="text-xl font-semibold mb-3"
          >
            Unique Products
          </h3>
          <p
            class="text-gray-600 text-sm"
          >
            Discover one-of-a-kind items you won't find in big box stores. Every product has a story.
          </p>
        </div>
        <div
          class="bg-white dark:bg-gray-300 p-8 rounded-lg shadow-sm text-center"
        >
          <div
            class="bg-blue-50 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-5 text-blue-600 text-2xl"
          >
            ü§ù
          </div>
          <h3
            class="text-xl font-semibold mb-3"
          >
            Support Local
          </h3>
          <p
            class="text-gray-600 text-sm"
          >
            Your purchases directly support independent creators and small business owners in your community.
          </p>
        </div>
        <div
          class="bg-white dark:bg-gray-300 p-8 rounded-lg shadow-sm text-center"
        >
          <div
            class="bg-blue-50 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-5 text-blue-600 text-2xl"
          >
            ‚≠ê
          </div>
          <h3
            class="text-xl font-semibold mb-3"
          >
            Verified Quality
          </h3>
          <p
            class="text-gray-600 text-sm"
          >
            Every seller is vetted for quality and service. Shop with confidence knowing we stand behind them.
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
